pkg/route/routeapihelpers/routeapihelpers.go

```diff
@@ -42,3 +44,29 @@ func IngressURI(route *routev1.Route, host string) (*url.URL, *routev1.RouteIngr
 	}
 	return nil, nil, fmt.Errorf("no ingress for host %s in route %s in namespace %s", host, route.ObjectMeta.Name, route.ObjectMeta.Namespace)
 }
+
+// ValidateHost checks that a route's host name satisfies DNS requirements.
+// Unless the allowNonCompliant annotation is set to true, host name must have
+// at least two labels, with each label less than 63 chars from a given character
+// set, and total length less than 253 chars.  Trailing dots are allowed.
```

Comment:
What does "from a given character set" mean?  It needs to be alphanumeric (plus the hyphen) ASCII, right?

Reply:
As implemented by `IsDNS1123Subdomain`, the character set is `lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character`.  I'll add it to the comment.